resolver/resolver.go

```diff
@@ -203,6 +203,15 @@ type State struct {
 // gRPC to add new methods to this interface.
 type ClientConn interface {
 	// UpdateState updates the state of the ClientConn appropriately.
+	//
+	// If an error is returned, the resolver should try to resolve the
+	// target again. Ideally the resolver uses a backoff timer to prevent
+	// overloading the server with requests.
+	// If a resolver is certain that reresolving does not change the result,
+	// e.g. because it is a watch-based resolver, returned errors can be
+	// ignored.
+	// If the resolved State is the same then the last reported one, calling
+	// UpdateState can be omitted.
```

Comment:
The formatting is a little off, here.

- If they are just multiple sentences in a single paragraph, they should be together without any newlines except at 80 columns.
- If these are separate paragraphs, they should be separated by a blank (`//`) line.
- If they are bullet points, they should start with `//   -` (see "Lists" at https://tip.golang.org/doc/comment).

Reply:
I removed the newline after "overloading the server with requests."  and added a paragraph before "If the resolved State is the same "